source/common/upstream/upstream_impl.cc

```diff
@@ -84,6 +85,13 @@ ClusterInfoImpl::ClusterInfoImpl(const Json::Object& config, Runtime::Loader& ru
     lb_type_ = LoadBalancerType::Random;
   } else if (string_lb_type == "ring_hash") {
     lb_type_ = LoadBalancerType::RingHash;
+  } else if (string_lb_type == "original_dst_lb") {
+    if (config.getString("type") != "original_dst") {
+      throw EnvoyException(
```

Comment:
I realize we have been bad about doing this in many places, but please test with EXPECT_THROW_WITH_MESSAGE for better tests.